### ISSUE TITLE
`dampening` and `weight_decay` parameter should be explained in order

### ISSUE THREAD

**Issue by hyperkai** [pytorch/pytorch#137356]
### 📚 The doc issue

[The doc](https://pytorch.org/docs/stable/generated/torch.optim.SGD.html) of `optim.SGD()` shows `dampening` and `weight_decay` parameter in order as shown below:

> Class torch.optim.SGD(..., momentum=0, dampening=0, weight_decay=0, nesterov=False, ...)

But in `Parameters`  section, `dampening` and `weight_decay` parameter are explained in opposite order as shown below:

> Parameters
> - ...
> - momentum ([float](https://docs.python.org/3/library/functions.html#float), optional) – momentum factor (default: 0)
> - weight_decay ([float](https://docs.python.org/3/library/functions.html#float), optional) – weight decay (L2 penalty) (default: 0) <- Here
> - dampening ([float](https://docs.python.org/3/library/functions.html#float), optional) – dampening for momentum (default: 0) <- Here
> - nesterov ([bool](https://docs.python.org/3/library/functions.html#bool), optional) – enables Nesterov momentum (default: False)   
> - ...

### Suggest a potential alternative/fix

So in `Parameters`  section, `dampening` and `weight_decay` parameter should be explained in order as shown below:

> Parameters
> - ...
> - momentum ([float](https://docs.python.org/3/library/functions.html#float), optional) – momentum factor (default: 0)
> - dampening ([float](https://docs.python.org/3/library/functions.html#float), optional) – dampening for momentum (default: 0) <- Here
> - weight_decay ([float](https://docs.python.org/3/library/functions.html#float), optional) – weight decay (L2 penalty) (default: 0) <- Here
> - nesterov ([bool](https://docs.python.org/3/library/functions.html#bool), optional) – enables Nesterov momentum (default: False)   
> - ...

cc @svekars @brycebortree @sekyondaMeta @vincentqb @jbschlosser @albanD @janeyx99 @crcrpar
